src/lsstseries/analysis/structurefunction2.py

```diff
@@ -10,9 +10,9 @@ def calc_sf2(
 
     Parameters
     ----------
-    lc_id : 'numpy.ndarray' (N,)
+    lc_id : `numpy.ndarray` (N,)
         Array of lightcurve ids per data point.
-    time : `numpy.ndarray` (N,) or None
+    time : `numpy.ndarray` (N,) or `None`
```

Comment:
Is this how it is implemented. You need to specify None for times? What happens when you pass only a single array of values?

Reply:
At present, yes, this is how it's implemented. A user would pass either `None` or an array of timestamps. Within the array of timestamps, some or all of the elements could be `None`. 
For instance, these are currently valid inputs:

- `None`
- `[None, None, None, None, ...]` 
- `[1, None, 4, 11, None, ...]`
- `[1, 2, 5, 11, ...]`


Perhaps I'm not understanding your question entirely. When you say "a single array of values", do you mean something like this:
` [1, 2, 5, 11, ...]` 
or something like this:
`[2, 2, 2, 2, ...]`

For the prior case, the function works as expected. For the later case, I believe the function would either fail or produce nonsensical output.

Reply:
Regarding first part -> I would want that an user can pass only one list of values, and that then we assume it is a list of fluxes. I guess that is a problem because argument `flux` is not the first argument, you can not specify that `id` and `time` are optional, and `flux` is not? I.e., I would prefer if an user can avoid having to specify `time = None`, and could just invoke `calc_sf2([1,2,3, ...])`.

Regarding second part -> In both cases the function should work. I believe that in the case `[2, 2, 2, 2, ...]` the output should be a list of zeros (because the variance of the function is zero) but it should.

Reply:
I believe I understand what you're saying now. Your preference is that the user would be able to invoke this function pass _just_ a list of fluxes and would assume equidistant times (and all errors = 0, etc...) If that assumption is correct, then yes, the function signature would either need to be refactored, or we would need to introduce an interface function on top that would allow that behavior. 

I'll put together a unit test to try out the [2,2,2,2...] scenario and see how the function responds. :)